cocos/core/data/object.ts

```diff
@@ -384,6 +384,9 @@ class CCObject implements EditorExtendableObject {
         }
 
         if (!EDITOR || legacyCC.GAME_VIEW) {
+            if (JSB &&  typeof (this.destruct) === 'function') {
```

Comment:
Why need to check whether it's a function? Is it possible not a function?

Reply:
Currently, it`s impossible, here we can check whether it is defined